### ISSUE TITLE
Plotting vectors using exodus II data

### ISSUE THREAD

**Issue by ahzeeshan** [yt-project/yt#2257]
### Bug report

**Bug summary**
Cannot plot velocity vectors (quivers) using data from exodus II file. The `ds.field_list` entry for the variable is `('all', nlc_x)` and `('all', 'nlc_y')` where `nlc_x` and `nlc_y` are the x and y components of the vector.

**Code for reproduction**

Link to the data file `test_101_other.e`: http://use.yt/upload/641ee3f8

```python
ds = yt.load("test_101_other.e", step=-1)
slc = yt.SlicePlot(ds, 2, ('all','eta'), origin="native")
slc.annotate_quiver(('all','nlc_x'), ('all','nlc_y'), factor = 16, plot_args={"color": "purple"})
```

**Actual outcome**

```
# If applicable, paste the console output here
#
#
yt : [WARNING  ] 2019-05-06 18:58:31,555 No info_records found
yt : [INFO     ] 2019-05-06 18:58:31,576 Loading coordinates
yt : [INFO     ] 2019-05-06 18:58:31,591 Loading connectivity
yt : [INFO     ] 2019-05-06 18:58:31,652 Parameters: current_time              = 0.09999999999999999
yt : [INFO     ] 2019-05-06 18:58:31,654 Parameters: domain_dimensions         = [1 1 1]
yt : [INFO     ] 2019-05-06 18:58:31,656 Parameters: domain_left_edge          = [-20. -20.   0.]
yt : [INFO     ] 2019-05-06 18:58:31,658 Parameters: domain_right_edge         = [220. 220.   1.]
yt : [INFO     ] 2019-05-06 18:58:31,660 Parameters: cosmological_simulation   = 0
yt : [INFO     ] 2019-05-06 18:58:31,662 Loading coordinates
yt : [INFO     ] 2019-05-06 18:58:31,677 Loading connectivity
yt : [INFO     ] 2019-05-06 18:58:31,909 xlim = -20.000000 220.000000
yt : [INFO     ] 2019-05-06 18:58:31,911 ylim = -20.000000 220.000000
yt : [INFO     ] 2019-05-06 18:58:31,914 xlim = -20.000000 220.000000
yt : [INFO     ] 2019-05-06 18:58:31,916 ylim = -20.000000 220.000000
yt : [INFO     ] 2019-05-06 18:58:31,918 Making a fixed resolution buffer of (('all', 'eta')) 800 by 800
---------------------------------------------------------------------------
YTPlotCallbackError                       Traceback (most recent call last)
/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    336             method = get_real_method(obj, self.print_method)
    337             if method is not None:
--> 338                 return method()
    339             return None
    340         else:

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_container.pyc in newfunc(*args, **kwargs)
     90             # it is the responsibility of _setup_plots to
     91             # call args[0].run_callbacks()
---> 92             args[0]._setup_plots()
     93         rv = f(*args, **kwargs)
     94         return rv

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_window.pyc in _setup_plots(self)
    958 
    959         self._set_font_properties()
--> 960         self.run_callbacks()
    961         self._plot_valid = True
    962 

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_window.pyc in run_callbacks(self)
   1015                     six.reraise(YTPlotCallbackError,
   1016                                 YTPlotCallbackError(callback._type_name, e),
-> 1017                                 sys.exc_info()[2])
   1018             for key in self.frb.keys():
   1019                 if key not in keys:

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_window.pyc in run_callbacks(self)
   1009                 callback = CallbackMaker(*args[1:], **kwargs)
   1010                 try:
-> 1011                     callback(cbw)
   1012                 except YTDataTypeUnsupported as e:
   1013                     six.reraise(YTDataTypeUnsupported, e)

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_modifications.pyc in _check_geometry(self, plot)
     69         cs = getattr(self, "coord_system", None)
     70         if supp is None or geom in supp:
---> 71             return func(self, plot)
     72         if cs in ("axis", "figure") and "force" not in supp:
     73             return func(self, plot)

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/visualization/plot_modifications.pyc in __call__(self, plot)
    454                                   fv_x,
    455                                   (x0, x1, y0, y1), 0, # bounds, antialias
--> 456                                   (period_x, period_y), periodic)
    457         pixelize_cartesian(pixY, plot.data['px'], plot.data['py'],
    458                                   plot.data['pdx'], plot.data['pdy'],

/Users/zeeshan/miniconda2/envs/atomic/lib/python2.7/site-packages/yt/utilities/lib/pixelization_routines.pyx in yt.utilities.lib.pixelization_routines.pixelize_cartesian()
     62                        np.float64_t[:] pdx,
     63                        np.float64_t[:] pdy,
---> 64                        np.float64_t[:] data,
     65                        bounds,
     66                        int antialias = 1,

YTPlotCallbackError: annotate_quiver callback failed with the following error: Buffer has wrong number of dimensions (expected 1, got 2)

<yt.visualization.plot_window.AxisAlignedSlicePlot at 0x11d0f8710>
```

**Expected outcome**

A plot showing the vector `nlc` using the x and y components `nlc_x` and `nlc_y` as shown in the example here: http://yt-project.org/doc/visualizing/callbacks.html#annotate_quiver

**Version Information**

  * Operating System: macOS Mojave version 10.14.4
  * Python Version: Python 2.7.15
  * yt version: 3.5.1 installed using pip
  * Other Libraries (if applicable):

**Comment by MatthewTurk** [comment]
@ahzeeshan, thanks to debugging by @Xarthisius and @munkm we figured out that this has been fixed already in master, thanks to #2105 from @jisuoqing !  Can you try updating it?

**Comment by ahzeeshan** [comment]
This worked, thanks.
